tests/cupy_tests/creation_tests/test_ranges.py

```diff
@@ -145,6 +145,34 @@ def test_linspace_float_underflow(self, xp):
             x /= 2
         return xp.linspace(0., x, 10, dtype=float)
 
+    @testing.with_requires('numpy>=1.16')
+    @testing.numpy_cupy_array_equal()
+    def test_linspace_array_start_stop(self, xp):
+        start = xp.array([-120, 120], dtype="int8")
+        stop = xp.array([100, -100], dtype="int8")
+        return xp.linspace(start, stop, num=50, dtype=float)
+
+    @testing.with_requires('numpy>=1.16')
+    @testing.numpy_cupy_array_equal()
+    def test_linspace_mixed_start_stop(self, xp):
+        start = 0.0
+        stop = xp.array([100, -100], dtype="int8")
```

Comment:
Could you parameterize dtype with `for_all_dtypes` or `for_all_dtypes_combination`? This test seems to fail if float32 array is given as the input.

Reply:
Okay, I have updated with more comprehensive testing.

There was indeed a subtle internal difference in the in dtypes being computed internally due to  #3035

If #3035 is addressed, then the refactoring done in 5dc9b9d should not be necessary.